File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/amap_to_am.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n, num_keys;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so dinh co trong AMap: ";
    cin >> num_keys;

    // Khoi tao ma tran n x n voi 0
    vector<vector<int>> adjMatrix(n, vector<int>(n, 0));

    cout << "Nhap AMap (u, so dinh ke, cac cap v w):\n";
    for (int i = 0; i < num_keys; ++i) {
        int u, k;
        cin >> u >> k;
        for (int j = 0; j < k; ++j) {
            int v, w;
            cin >> v >> w;
            adjMatrix[u][v] = w; // Gan w cho canh tu u den v
        }
    }

    cout << "Ma tran ke (AM):\n";
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            cout << adjMatrix[i][j] << " ";
        }
        cout << "\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/amap_to_am.py
n = int(input("Nhap so dinh: "))
num_keys = int(input("Nhap so dinh co trong AMap: "))

# Khoi tao ma tran ke n x n voi toan 0
adj_matrix = [[0 for _ in range(n)] for _ in range(n)]

print("Nhap AMap (u, so dinh ke, cac cap v w):")
for _ in range(num_keys):
    parts = input().split()
    u = int(parts[0])
    k = int(parts[1])
    for i in range(k):
        v = int(parts[2 + 2 * i])
        w = int(parts[3 + 2 * i])
        adj_matrix[u][v] = w  # Gan trong so w cho canh tu u den v

# In ma tran ke
print("Ma tran ke (AM):")
for row in adj_matrix:
    print(" ".join(map(str, row)))